### PR TITLE
Remove case sensitivity for string built-in in icon component

### PR DESCRIPTION
**Pull request by elyanah-aco**
### All Submissions:

- [x] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/pynecone-io/pynecone/blob/main/CONTRIBUTING.md) file?
- [x] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/pynecone-io/pynecone/pulls ) for the desired changed?

### Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### New Feature Submission:

- [x] Does your submission pass the tests? 
- [x] Have you linted your code locally prior to submission?

### Changes To Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### **After** these steps, you're ready to open a pull request.

PR removes case sensitivity for the `tag` argument in `pc.Icon` as requested in #420. Admittedly, this fix doesn't feel 'rigorous' enough for me and may need to be updated if future icons are added, so feel free to make suggestions.

### CODE REVIEW

File: pynecone/components/media/icon.py [hunk 1]
```diff
@@ -36,5 +38,30 @@ def create(cls, *children, **props):
             )
         if "tag" not in props.keys():
             raise AttributeError("Missing 'tag' keyword-argument for Icon")
+        else:
+            tag_keywords = [
```
Comment (FHU-yezi) [inline pynecone/components/media/icon.py:42]: May use tuple instead of list to avoid change this by accident?

Comment (picklelo) [reply pynecone/components/media/icon.py:42]: I'm a bit confused - what are the `tag_keywords` used for? Shouldn't the `.title()` call be enough?

Comment (elyanah-aco) [reply pynecone/components/media/icon.py:42]: No, that isn't enough for "multiword" tags (eg. `ExternalLinkIcon`, `RepeatClockIcon`). From what I found in testing, `tag` should still match actual Chakra icon name case to work

Comment (picklelo) [reply pynecone/components/media/icon.py:42]: Ah I see - maybe in those cases the user can specify the tag in `snake_case` like `pc.icon(tag="repeat_clock_icon")` and we can use the `utils.to_title_case` function in `utils.py`. What do you think?

Comment (elyanah-aco) [reply pynecone/components/media/icon.py:42]: Yes, that sounds better and more futureproof. We just need to make it clear that the user has to use `snake_case` in documentation

**Comment by picklelo** [comment]
I think something got messed up on the rebase there's a lot of extra changes - could you try again to rebase on `main`?

**Comment by elyanah-aco** [comment]
Will create new PR for this

**Pull request closed without merging**